lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3442,12 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+Also, cleans up temporary files for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
+``fetch``, ``stage``, or ``install`` are run again after this, Spack's
+build process will start from scratch.
```

Comment:
The functionality didn't change at all. The command:
```
$ spack clean --stage foo
```
in this branch is equivalent to:
```console
$ spack clean foo
$ spack purge --stage
```
in `develop` (so it deletes the stage for foo, and then it deletes all stages.). Same for other commands. This is reflected in the command help (hopefully):
```
$ spack help clean
usage: spack clean [-h] [-s] [-d] [-m] [-a] ...

remove temporary build files and/or downloaded archives

positional arguments:
  specs             removes the build stages and tarballs for specs

optional arguments:
  -h, --help        show this help message and exit
  -s, --stage       remove all temporary build stages (default)
  -d, --downloads   remove cached downloads
  -m, --misc-cache  remove long-lived caches, like the virtual package index
  -a, --all         equivalent to -sdm
```

----

Good catch for the docs. I'll update the reference.